scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -309,154 +309,228 @@ For this example we will focus on identifying DE genes in these `Tumor_Myoblast`
+
+As a reminder, the uncorrected gene expression data, found in the `counts` and `logcounts` assays, correspond to data that has been merged (the first step we walked through prior to integration) into the same SCE but not yet integrated.
+`DESeq2` expects raw counts as input so we will be using data found in the `counts` assay.
+
 ### Pseudo-bulking 
 
-Before we can compare the gene expression profiles of myoblasts in ARMS vs. ERMS patients, we will need to "pseudo-bulk" the gene counts. 
-Pseudo-bulking creates a new counts matrix that contains the total counts across all cells of a given label (e.g. cell type) for each sample. 
-This allows us to use single-cell resolution to define the labels and sum gene counts across groups of cells containing the same label, and avoids counting each cell as its own replicate. 
+Before we can compare the gene expression profiles of myoblasts in ARMS vs. ERMS samples, we will need to "pseudo-bulk" the gene counts. 
+Pseudo-bulking creates a new counts matrix that contains the sum of the counts from all cells with a given label (e.g., cell type) for each sample ([Tung _et al._, 2017](https://doi.org/10.1038/srep39921)). 
+If we were to keep each cell's counts separate, treating them as replicate samples, we would end up with inflated statistics. 
+By pseudo-bulking first, will now have one count for each gene for each sample and we can take advantage of well-established methods for differential expression with bulk RNA-seq.
 
 Pseudo-bulking is implemented prior to differential expression analysis on single-cell data for the following reasons: 
 
-- Produces larger counts, which allows us to use standard normalization and differential expression methods used by bulk RNA-sequencing. 
--  Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
+- Produces larger and less sparse counts, which allows us to use standard normalization and differential expression methods used by bulk RNA-seq. 
+- Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
 - Masks variance within a sample to emphasize variance across samples.
-
-We are starting with the simple comparison of looking at one cell type and comparing across two groups of samples: the ERMS and ARMS subdiagnoses. 
-Our first step will be to pseudo-bulk our dataset to group our cells by cell type and by sample.
-This will create a new `SingleCellExperiment` (SCE) object that contains the pseudo-bulked counts for all genes. 
-We can subset this SCE to just contain our cell type of interest (tumor myoblasts) for input to differential expression analysis. 
+This can be both good and bad! 
+Genes that are homogeneous across a sample are more likely to be identified as differentially expressed, while genes with high intra-sample heterogeneity will have a weaker signal.
```

Comment:
Yes your example totally makes sense! I was thinking more along the lines if you have a gene that's uniformly up or down across cells then generally that would result in a stronger signal, whereas variation in your sample, even if just a small amount, will result in a weaker signal. Taking this from some of the explanation in the third bullet under the pseudo-bulking section of OSCA: http://bioconductor.org/books/3.16/OSCA.multisample/multi-sample-comparisons.html#creating-pseudo-bulk-samples

I'll update this to be more clear about what is meant by homogeneous.

Reply:
I ended up removing the comment about homogeneous genes and reworded to emphasize that you would miss genes with intra-sample heterogeneous in https://github.com/AlexsLemonade/training-modules/pull/600/commits/ee2e73be65cf5a9fb1dd99f94bd9eed5fe773485 (and added in the missing chunk labels there too). Let me know what you think.

Reply:
I kind of partially disagree with part of their statement here, which is the point I was trying to get across: 

> Masking is generally desirable as DEGs - unlike marker genes - do not need to have low within-sample variance to be interesting, e.g., if the treatment effect is consistent across replicate populations but heterogeneous on a per-cell basis.

The part I disagree with is the implicit assumption that not knowing about heterogeneous cell expression is just fine. I kind of think it is worth knowing, and a model that ignores it seems weaker to me. But this is really subtle, and probably not worth the effort to try to get across in a short sentence. 

Maybe something like this? Or maybe just ignore it... This might be a thing for a slide where illustration of the point is easier.

```
Masking intra-sample variation means you might miss identifying genes where average expression doesn't change between samples but the degree of cell-to-cell variation does.
```